backends/advanced/src/advanced_omi_backend/services/memory/providers/llm_providers.py

```
@@ -285,20 +301,21 @@ async def generate_embeddings(self, texts: List[str]) -> List[List[float]]:
-            try:
+            # Add 10-second timeout to prevent hanging on API calls
+            async with asyncio.timeout(10):
                 client = _get_openai_client(api_key=self.api_key, base_url=self.base_url, is_async=True)
                 await client.models.list()
                 return True
-            except Exception as e:
-                memory_logger.error(f"OpenAI connection test failed: {e}")
-                return False
-            
+        except asyncio.TimeoutError:
+            memory_logger.error(f"OpenAI connection test timed out after 10s - check network connectivity and API endpoint")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)